### PR TITLE
Add muon benchmarks

### PR DESCRIPTION
**Pull request by sebouh137**
### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### CODE REVIEW

File: benchmarks/insert_muon/config.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+simulate:insert_muon:
+  stage: simulate
+  extends: .phy_benchmark
```
Comment (veprbl) [inline benchmarks/insert_muon/config.yml:3]: https://eicweb.phy.anl.gov/EIC/benchmarks/detector_benchmarks/-/pipelines/102788

File: benchmarks/zdc_muon/config.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+simulate:zdc_muon:
+  stage: simulate
+  extends: .phy_benchmark
```
Comment (veprbl) [inline benchmarks/zdc_muon/config.yml:3]: Same

File: benchmarks/insert_muon/config.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+simulate:insert_muon:
```
Comment (veprbl) [inline benchmarks/insert_muon/config.yml:1]:
```suggestion
sim:insert_muon:
```

File: benchmarks/insert_muon/config.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    - snakemake --cores 1 sim_output/insert_muon/epic_craterlake_rec_mu-_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:insert_muon:
+  stage: analyze
+  extends: .phy_benchmark
+  needs: ["simulate:insert_muon"]
```
Comment (veprbl) [inline benchmarks/insert_muon/config.yml:25]:
```suggestion
  needs: ["sim:insert_muon"]
```

File: benchmarks/zdc_muon/config.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+simulate:zdc_muon:
```
Comment (veprbl) [inline benchmarks/zdc_muon/config.yml:1]:
```suggestion
sim:zdc_muon:
```

File: benchmarks/zdc_muon/config.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    - snakemake --cores 1 sim_output/zdc_muon/epic_craterlake_rec_mu-_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:zdc_muon:
+  stage: analyze
+  extends: .phy_benchmark
+  needs: ["simulate:zdc_muon"]
```
Comment (veprbl) [inline benchmarks/zdc_muon/config.yml:25]:
```suggestion
  needs: ["sim:zdc_muon"]
```

File: benchmarks/zdc_muon/config.yml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  script:
+    - snakemake --cores 1 sim_output/zdc_muon/epic_craterlake_rec_mu-_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:zdc_muon:
+  stage: analyze
+  extends: .phy_benchmark
```
Comment (veprbl) [inline benchmarks/zdc_muon/config.yml:24]:
```suggestion
  extends: .det_benchmark
```

File: benchmarks/insert_muon/config.yml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  script:
+    - snakemake --cores 1 sim_output/insert_muon/epic_craterlake_rec_mu-_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:insert_muon:
+  stage: analyze
+  extends: .phy_benchmark
```
Comment (veprbl) [inline benchmarks/insert_muon/config.yml:24]:
```suggestion
  extends: .det_benchmark
```

File: benchmarks/insert_muon/config.yml (hunk 1 above)
Comment (veprbl) [reply benchmarks/insert_muon/config.yml:3]:
```suggestion
  extends: .det_benchmark
```

File: benchmarks/zdc_muon/config.yml (hunk 2 above)
Comment (veprbl) [reply benchmarks/zdc_muon/config.yml:3]:
```suggestion
  extends: .det_benchmark
```

File: benchmarks/insert_muon/config.yml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+analyze:insert_muon:
+  stage: analyze
+  extends: .det_benchmark
+  needs: ["sim:insert_muon"]
+  script:
+    - mkdir -p results/epic_craterlake
+    - python benchmarks/insert_muon/analysis/muon_plots.py results/epic_craterlake/insert_muon
+
+collect_results:insert_muon:
+  stage: collect_results
```
Comment (veprbl) [inline benchmarks/insert_muon/config.yml:31]:
```suggestion
  stage: collect
```

File: benchmarks/zdc_muon/config.yml [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+analyze:zdc_muon:
+  stage: analyze
+  extends: .det_benchmark
+  needs: ["sim:zdc_muon"]
+  script:
+    - mkdir -p results/epic_craterlake
+    - python benchmarks/zdc_muon/analysis/muon_plots.py results/epic_craterlake/zdc_muon
+
+collect_results:zdc_muon:
+  stage: collect_results
```
Comment (veprbl) [inline benchmarks/zdc_muon/config.yml:31]:
```suggestion
  stage: collect
```

File: benchmarks/zdc_muon/config.yml [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+  timeout: 1 hours
+  script:
+    - snakemake --cores 1 sim_output/zdc_muon/epic_craterlake_rec_mu-_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:zdc_muon:
+  stage: analyze
```
Comment (veprbl) [inline benchmarks/zdc_muon/config.yml:23]:
```suggestion
  stage: benchmark
```

File: benchmarks/insert_muon/config.yml [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+  timeout: 1 hours
+  script:
+    - snakemake --cores 1 sim_output/insert_muon/epic_craterlake_rec_mu-_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:insert_muon:
+  stage: analyze
```
Comment (veprbl) [inline benchmarks/insert_muon/config.yml:23]:
```suggestion
  stage: benchmark
```

File: .gitlab-ci.yml [hunk 13]
```diff
@@ -44,6 +44,7 @@ stages:
   - simulate
   - calibrate
   - benchmarks
+  - analyze
```
Comment (veprbl) [inline .gitlab-ci.yml:47]:
```suggestion
```

File: benchmarks/insert_muon/config.yml [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+      - P: 80
+  timeout: 1 hours
+  script:
+    - snakemake --cores 1 sim_output/insert_muon/epic_craterlake_rec_mu-_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:insert_muon:
```
Comment (veprbl) [inline benchmarks/insert_muon/config.yml:22]:
```suggestion
bench:insert_muon:
```

File: benchmarks/zdc_muon/config.yml [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+      - P: 80
+  timeout: 1 hours
+  script:
+    - snakemake --cores 1 sim_output/zdc_muon/epic_craterlake_rec_mu-_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:zdc_muon:
```
Comment (veprbl) [inline benchmarks/zdc_muon/config.yml:22]:
```suggestion
bench:zdc_muon:
```

File: benchmarks/insert_muon/config.yml [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+  stage: benchmarks
+  extends: .det_benchmark
+  needs: ["sim:insert_muon"]
+  script:
+    - mkdir -p results/epic_craterlake
+    - python benchmarks/insert_muon/analysis/muon_plots.py results/epic_craterlake/insert_muon
+
+collect_results:insert_muon:
+  stage: collect
+  needs: ["analyze:insert_muon"]
```
Comment (veprbl) [inline benchmarks/insert_muon/config.yml:32]:
```suggestion
  needs: ["bench:insert_muon"]
```

File: benchmarks/zdc_muon/config.yml [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+  stage: benchmarks
+  extends: .det_benchmark
+  needs: ["sim:zdc_muon"]
+  script:
+    - mkdir -p results/epic_craterlake
+    - python benchmarks/zdc_muon/analysis/muon_plots.py results/epic_craterlake/zdc_muon
+
+collect_results:zdc_muon:
+  stage: collect
+  needs: ["analyze:zdc_muon"]
```
Comment (veprbl) [inline benchmarks/zdc_muon/config.yml:32]:
```suggestion
  needs: ["bench:zdc_muon"]
```

**Comment by sebouh137** [comment]
@wdconinc Is there a way to specify that I want to turn off the magnetic field for the ZDC simulation?  I want to use a standard configuration (for instance craterlake) minus the magnetic field (so that I can send muons to the ZDC from the target and use them to check the acceptance, without them being swept away by a magnetic field).  Or do I just need to create a new configuration in epic for this?

**Pull request merged**